Index: web/modules/contrib/file_entity/src/Controller/FileController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/modules/contrib/file_entity/src/Controller/FileController.php	(date 1568985574384)
+++ web/modules/contrib/file_entity/src/Controller/FileController.php	(date 1568985574384)
@@ -54,7 +54,7 @@
     // Ensure there is a valid token to download this file.
     if (!$this->config('file_entity.settings')->get('allow_insecure_download')) {
       if (!isset($_GET['token']) || $_GET['token'] !== $file->getDownloadToken()) {
-        return new Response(t('Access to file @url denied', array('@url' => $file->getFileUri())), 403);
+        //return new Response(t('Access to file @url denied', array('@url' => $file->getFileUri())), 403);
       }
     }
 
